back/tests/cmd/openbsrv-integration/integration_test.go

```diff
@@ -0,0 +1,445 @@
+
+	// User Service Tests
+	userClnt := pb.NewUserSvcClient(conn)
+	t.Run("Add and find user role", userSvcAddAndFndRoleFunc(ctx, conn, userClnt))
+	t.Run("Add and find users", userSvcAddAndFndUsersFunc(ctx, conn, userClnt))
+	t.Run("Find and delete user", userSvcDelUserFunc(ctx, conn, userClnt))
+
+	// Post Service Tests
+	postClnt := pb.NewPostClient(conn)
+	t.Run("Add type", postSvcAddAndFndTypesFunc(ctx, conn, postClnt))
+	t.Run("Add and find posts", postSvcAddAndFndPostsFunc(ctx, conn, postClnt))
+	t.Run("Add, find and edit post", postSvcEdtPostFunc(ctx, conn, postClnt))
+	t.Run("Find and delete post", postSvcDelPostFunc(ctx, conn, postClnt))
+}
+
+func userSvcAddAndFndRoleFunc(ctx context.Context, conn *grpc.ClientConn, clnt pb.UserSvcClient) func(*testing.T) {
+	return func(t *testing.T) {
+		want := "testRole"
+
+		_, err := clnt.AddRole(ctx, &pb.AddRoleReq{Name: want})
+		if err != nil {
+			t.Log(err)
+			t.Fail()
+		}
+
+		r, err := clnt.FndRoles(ctx, &pb.FndRolesReq{
+			RoleIds:   []string{},
+			RoleNames: []string{want},
+			Limit:     100,
+			Lapse:     0,
+		})
+
+		if len(r.Items) != 1 || r.Items[0].Name != want {
+			t.Logf("got: %v, want: %s", r, want)
+			t.Fail()
+		}
+	}
+}
+
+type user struct {
+	Username    string
+	Email       string
+	EmailHold   bool
+	Altmail     string
+	AltmailHold bool
+	FullName    string
+	Avatar      string
+}
+
+func userSvcAddAndFndUsersFunc(ctx context.Context, conn *grpc.ClientConn, clnt pb.UserSvcClient) func(*testing.T) {
+	return func(t *testing.T) {
+		tests := []struct {
+			addReq *pb.AddUserReq
+			want   user
+			fndReq pb.FndUsersReq
+		}{
+			{
+				&pb.AddUserReq{
+					Username:    "test username A",
+					Email:       "user_a@email.com",
+					EmailHold:   false,
+					Altmail:     "",
+					AltmailHold: false,
+					FullName:    "test user full name A",
+					Avatar:      "test user avatar A",
+					Password:    "test user password A",
+				},
+				user{
+					Username:    "test username A",
+					Email:       "user_a@email.com",
+					EmailHold:   false,
+					Altmail:     "",
+					AltmailHold: false,
+					FullName:    "test user full name A",
+					Avatar:      "test user avatar A",
+				},
+				pb.FndUsersReq{
+					RoleIds:     []string{},
+					Email:       "user_a@email.com",
+					EmailHold:   false,
+					Altmail:     "",
+					AltmailHold: false,
+					Limit:       100,
+					Lapse:       0,
+				},
+			},
+			{
+				&pb.AddUserReq{
+					Username:    "test username B",
+					Email:       "userB@email.com",
+					EmailHold:   false,
+					Altmail:     "",
+					AltmailHold: false,
+					FullName:    "test user full name B",
+					Avatar:      "test user avatar B",
+					Password:    "test user password B",
+				},
+				user{
+					Username:    "test username B",
+					Email:       "userB@email.com",
+					EmailHold:   false,
+					Altmail:     "",
+					AltmailHold: false,
+					FullName:    "test user full name B",
+					Avatar:      "test user avatar B",
+				},
```

Comment:
I think this type might need to be &pb.AddUserResp. Please remove `user` if it's unneeded.

Reply:
Done